src/ui/ui_exports.js

```diff
@@ -48,15 +49,21 @@ class UiExports {
       case 'app':
       case 'apps':
         return (plugin, specs) => {
+          const id = plugin.id;
           for (let spec of [].concat(specs || [])) {
-            let app = this.apps.new(_.defaults({}, spec, {
-              id: plugin.id,
-              urlBasePath: this.urlBasePath
-            }));
+            const app = this.apps.new({ id, ...spec });
             plugin.apps.add(app);
           }
         };
 
+      case 'link':
+      case 'links':
+        return (plugin, spec) => {
+          for (const spec of [].concat(spec || [])) {
```

Comment:
Woah, `const` is applied within the loop block rather than in the same parent context as the loop?  I would have assumed that this would fail on >1 iterations.